src/modules/75-cf/pages/feature-flags/components/__tests__/NoFeatureFlags.test.tsx

```diff
@@ -0,0 +1,122 @@
+        data: mockEnvironments,
+        loading: false,
+        error: undefined,
+        refetch: jest.fn()
+      })
+    })
+  })
+
+  afterAll(() => {
+    jest.resetAllMocks()
+  })
+
+  test('It should render "No Flags" image, message & New Flag button when the project has no feature flags', async () => {
+    renderComponent({ hasFeatureFlags: false, hasFlagFilter: false, hasSearchTerm: false })
+
+    await waitFor(() => {
+      expect(screen.getByTestId('nodata-image')).toBeVisible()
+      expect(screen.getByText('cf.noFlag')).toBeVisible()
+      expect(screen.getByText('cf.featureFlags.newFlag')).toBeVisible()
+    })
+
+    userEvent.click(screen.getByText('cf.featureFlags.newFlag'))
+
+    // Check New Flag modal is open
+    await waitFor(() => {
+      expect(screen.getByText('cf.featureFlags.typeOfFlag')).toBeVisible()
+    })
+  })
+
+  test('It should render "No Results" image, filter criteria message & Clear Filters button when no existing feature flags match the filter criteria', async () => {
+    renderComponent({ hasFeatureFlags: true, hasFlagFilter: true, hasSearchTerm: false })
+
+    await waitFor(() => {
+      expect(screen.getByTestId('nodata-image')).toBeVisible()
+      expect(screen.getByText('common.filters.noMatchingFilterData')).toBeVisible()
+      expect(screen.getByText('cf.featureFlags.changeOrReset')).toBeVisible()
+      expect(screen.getByText('cf.featureFlags.resetFilters')).toBeVisible()
+    })
+
+    userEvent.click(screen.getByText('cf.featureFlags.resetFilters'))
+
+    await waitFor(() => {
+      expect(onClearFilter).toHaveBeenCalled()
+    })
+  })
+
+  test('It should render "No Results" image, search criteria message & Clear Search button when no existing feature flags match the search criteria', async () => {
+    renderComponent({ hasFeatureFlags: true, hasFlagFilter: false, hasSearchTerm: true })
+
+    await waitFor(() => {
+      expect(screen.getByTestId('nodata-image')).toBeVisible()
+      expect(screen.getByText('cf.noResultMatch')).toBeVisible()
+      expect(screen.getByText('cf.featureFlags.clearSearch')).toBeVisible()
+    })
+
+    await act(async () => {
+      userEvent.click(screen.getByText('cf.featureFlags.clearSearch'))
+    })
+    expect(onClearSearch).toHaveBeenCalled()
+  })
+
+  test('It should render the filter message when both flagFilter and searchTerm are present', async () => {
+    renderComponent({ hasFeatureFlags: true, hasFlagFilter: true, hasSearchTerm: true })
+
+    await waitFor(() => {
+      expect(screen.getByTestId('nodata-image')).toBeVisible()
+      expect(screen.getByText('common.filters.noMatchingFilterData')).toBeVisible()
+      expect(screen.getByText('cf.featureFlags.changeOrReset')).toBeVisible()
+      expect(screen.getByText('cf.featureFlags.resetFilters')).toBeVisible()
+    })
+
+    await act(async () => {
+      userEvent.click(screen.getByText('cf.featureFlags.resetFilters'))
+    })
+    expect(onClearFilter).toHaveBeenCalled()
```

Comment:
You don't need the `act`, but you might need a `waitFor` around your assertion
```suggestion
    userEvent.click(screen.getByText('cf.featureFlags.resetFilters'))

    await waitFor(() => expect(onClearFilter).toHaveBeenCalled())
```